plugins/woocommerce/tests/php/includes/class-wc-install-test.php

```diff
@@ -164,4 +164,22 @@ public function test_delete_obsolete_notes_deletes_only_selected_notes() {
 		$data_store->delete( $note_2 );
 	}
 
+	/**
+	 * Test that maybe_set_store_id only sets an ID when it isn't already present.
+	 */
+	public function test_maybe_set_store_id() {
+
+		// simulate a store ID not being set.
+		delete_option( \WC_Install::STORE_ID_OPTION );
+		$store_id = \WC_Install::maybe_set_store_id();
+		$this->assertSame( $store_id, get_option( \WC_Install::STORE_ID_OPTION ) );
+		// uuid4 is 36 characters long.
+		$this->assertSame( 36, strlen( $store_id ) );
+
+		// simulate a store ID already being set.
+		$existing_store_id = \WC_Install::maybe_set_store_id();
+		$this->assertSame( $store_id, $existing_store_id );
+		// cleanup.
+		delete_option( \WC_Install::STORE_ID_OPTION );
```

Comment:
Got it! I updated the tests to not rely on `maybe_set_store_id` returning a value and removed the check for whether the ID is in fact the same. I kept the check for 36 characters, though, because I want to ensure that the id is a valid `uuid4` string.

The second half of the test checks the state when the `store_id` has already been set.